### DISCUSSION TITLE
Add objects palette

### DISCUSSION THREAD

**Discussion by abramenko95** [maxGraph/maxGraph#600] [Q&A]
I have custom objects palette in XML file. Object image is SVG incoded to base64.
`<shape aspect="variable" w="130" h="86" name="example">
	<foreground>
		<image w="130" h="86" y="0" x="0" src="data:image/svg+xml;base64,..."/>
	</foreground>
</shape>`
I need add my custom objects palette to diagram like in this mxGraph example or another way.
https://github.com/jgraph/mxgraph/blob/ff141aab158417bd866e2dfebd06c61d40773cd2/javascript/examples/grapheditor/www/js/Sidebar.js#L4309
I figured out how to add base64 image to toolbar or how to register shape as some SVG like functions, but I need add base64 image to diagram.
https://github.com/maxGraph/maxGraph/blob/62d4c8f1e45b2480a5f8466c8320e6280f010903/packages/html/stories/Toolbar.stories.ts#L142
https://github.com/maxGraph/maxGraph/blob/62d4c8f1e45b2480a5f8466c8320e6280f010903/packages/html/stories/Stencils.stories.ts#L133

**Comment by tbouffard** [comment]
Hi, I guess that you could tryo to use base64 content for the image, by calling `MaxToolbar.addItem`: https://github.com/maxGraph/maxGraph/blob/62d4c8f1e45b2480a5f8466c8320e6280f010903/packages/core/src/gui/MaxToolbar.ts#L102
The icon parameter is used to set the img.src attribute, so it could work when passing a base64 value.
See: https://github.com/maxGraph/maxGraph/blob/62d4c8f1e45b2480a5f8466c8320e6280f010903/packages/core/src/gui/MaxToolbar.ts#L115-L117

Did you try this? I didn't try it myself.
